packages/libs/src/sdk/api/generated/full/models/AccessGate.ts

```diff
@@ -0,0 +1,86 @@
+/* tslint:disable */
+/* eslint-disable */
+// @ts-nocheck
+/**
+ * API
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import {
+    FollowGate,
+    instanceOfFollowGate,
+    FollowGateFromJSON,
+    FollowGateFromJSONTyped,
+    FollowGateToJSON,
+} from './FollowGate';
+import {
+    NftGate,
+    instanceOfNftGate,
+    NftGateFromJSON,
+    NftGateFromJSONTyped,
+    NftGateToJSON,
+} from './NftGate';
+import {
+    PurchaseGate,
+    instanceOfPurchaseGate,
+    PurchaseGateFromJSON,
+    PurchaseGateFromJSONTyped,
+    PurchaseGateToJSON,
+} from './PurchaseGate';
+import {
+    TipGate,
+    instanceOfTipGate,
+    TipGateFromJSON,
+    TipGateFromJSONTyped,
+    TipGateToJSON,
+} from './TipGate';
+
+/**
+ * @type AccessGate
+ * 
+ * @export
+ */
+export type AccessGate = FollowGate | NftGate | PurchaseGate | TipGate;
```

Comment:
This feels like it's going to be slightly cumbersome to work with on the client side. i.e. we may need to write helpers to make it easy to determine which type it is and wrap a typeguard around it (the checkers generated by OAPI return booleans and not typeguards, unfortunately)